packages/amplify-e2e-tests/src/categories/api.ts

```diff
@@ -76,10 +79,11 @@ export function addApiWithSchema(cwd: string, schemaFile: string) {
   });
 }
 
-export function addApiWithSchemaAndConflictDetection(cwd: string, schemaFile: string) {
+export function addApiWithSchemaAndConflictDetection(cwd: string, schemaFile: string, useLocalCLI: boolean = false) {
   const schemaPath = getSchemaPath(schemaFile);
+  const amplifySpawn = useLocalCLI ? amplify : getCLIPath();
```

Comment:
There are several other places that have the same pattern. But when running on CI, getCLIPath() will just return `amplify`, which means no matter the `useLocalCLI` is true or false, it is the same `amplify`, then there's no comparison between the existing published amplify and the  CLI codebase under test that is to be published.